File: backend/app/schemas/access_expiration.py
import uuid
from datetime import datetime
from pydantic import BaseModel


class AccessExpirationCreate(BaseModel):
    user_id: uuid.UUID
    expires_at: datetime


class AccessExpirationUpdate(BaseModel):
    expires_at: datetime


class AccessExpirationResponse(BaseModel):
    model_config = {"from_attributes": True}

    id: uuid.UUID
    user_id: uuid.UUID
    company_id: uuid.UUID
    expires_at: datetime
    is_notified: bool
    was_auto_blocked: bool
    created_at: datetime
    updated_at: datetime

File: frontend/src/types/access_expiration.ts
export interface AccessExpiration {
  id: string
  user_id: string
  company_id: string
  expires_at: string
  is_notified: boolean
  was_auto_blocked: boolean
  created_at: string
  updated_at: string
}

export interface AccessExpirationCreate {
  user_id: string
  expires_at: string
}

export interface AccessExpirationUpdate {
  expires_at: string
}
